0-intro-to-programming/slides-code/0-intro-to-programming.Rmd

```diff
@@ -0,0 +1,720 @@
+
+**Free/Open-source**: Python is and will always be 100% free to use
+
+**Easy syntax**: Python code reads more similarly to regular English
+
+**Widely used**: Many users means many user developed tools
+
+**#1 in data science**: The language with most data science tools
+
+**General purpose**: Python can be used for data science, servers, software etc.
+]
+
+.pull-right[
+  .center[
+    ```{r, eval=TRUE, echo=FALSE,out.width = '75%'}
+      knitr::include_graphics("img/pythonlogo.png")
+    ```
+  ]
+] 
+
+---
+class: center, middle
+
+# Questions? 
+
+---
+class: inverse, center, middle
+
+# How does one write good code?
+
+---
+class: middle
+
+.pull-left[
+  
+  ## Learn how to commnuicate data science well
+
+  **Correct code:** Code that generates correct results
+  
+  **Good code:** Code that can be understood and used by others
+  
+  Code should be **both** correct code and good code.
+  Only then is code useful as an output of your data work.
+  
+  Start by writing _correct code_, 
+  then keep going to write _good code_.
+]
+
+.pull-right[
+  .center[
+    ```{r, eval=TRUE, echo=FALSE,out.width = '80%'}
+      knitr::include_graphics("img/coms-well-by-code.png")
+    ```
+  ]
+] 
+
+
+---
+class: middle
+
+## Trial and error!
+
+**Writing correct code usually takes many errors**
+
+Don't be afraid of making errors, as code should not edit the original data! 
+
+Even the best coders use trial and error - 
+what makes them the best is how well they understand the errors
+
+---
+
+class: middle
+
+.pull-left[
+  .center[
+    ```{r, eval=TRUE, echo=FALSE,out.width = '100%'}
+      knitr::include_graphics("img/ihearterrors.png")
+    ```
+  ]
+] 
+
+.pull-right[
+
+## Errors and bugs
+
+* Errors is a part of programming and they should be your friend
+  
+* Python always give you some message when there is an error
+  
+* Learn how to interpret error messages - Google them
+
+* You will have thousands of errors and bugs - 
+test run your code and find out where it breaks
+]
+
+---
+class: middle
+
+.pull-left[
+  .center[
+   # When your code works<br>you are only half-done
+  ]
+  .right[
+    Ancient proverb
+  ]
+]
+
+.pull-right[
+
+  ### Path towards good code
+  
+  How to make correct code into good code?
+  
+  Ask yourself 
+  "_Would someone with your level of Python knowledge
+  be able to follow the code?_"
+  
+  * Simplify
+  * Peer-review
+  * Document
+  
+  Settling for "_just_" correct code does not 
+  save you or your team any time in the long run.
+
+]
+
+---
+
+class: middle
+
+## Re-use code
+
+We can easily re-use code within and across projects as code is separate from inputs and outputs. Three levels of re-using code:
+
+* **Copy-paste:** Copy someone else's code and modify it to fit your context
+
+* **Functions:** A block of code that can be referenced across a project
+
+* **Library:** Code packages developed by others to solve difficult tasks but made easy to use in any project
```

Comment:
s/Library/Libraries

To be consistent with above

Reply:
```suggestion
* **Libraries:** Code packages developed by others to solve difficult tasks but made easy to use in any project
```